qemu_v8.mk

```diff
@@ -0,0 +1,278 @@
+################################################################################
+# Following variables defines how the NS_USER (Non Secure User - Client
+# Application), NS_KERNEL (Non Secure Kernel), S_KERNEL (Secure Kernel) and
+# S_USER (Secure User - TA) are compiled
+################################################################################
+override COMPILE_NS_USER   := 64
+override COMPILE_NS_KERNEL := 64
+override COMPILE_S_USER    := 64
+override COMPILE_S_KERNEL  := 64
+
+-include common.mk
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+ARM_TF_PATH			?= $(ROOT)/arm-trusted-firmware
+
+EDK2_PATH			?= $(ROOT)/edk2
+EDK2_BIN			?= $(EDK2_PATH)/QEMU_EFI.fd
+
+QEMU_PATH			?= $(ROOT)/qemu
+
+SOC_TERM_PATH			?= $(ROOT)/soc_term
+STRACE_PATH			?= $(ROOT)/strace
+
+DEBUG = 1
+
+################################################################################
+# Targets
+################################################################################
+all: arm-tf qemu soc-term linux strace update_rootfs
+all-clean: arm-tf-clean busybox-clean linux-clean \
+	optee-os-clean optee-client-clean qemu-clean \
+	soc-term-clean check-clean strace-clean
+
+-include toolchain.mk
+
+################################################################################
+# ARM Trusted Firmware
+################################################################################
+ARM_TF_EXPORTS ?= \
+	CFLAGS="-O0 -gdwarf-2" \
+	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+
+ARM_TF_FLAGS ?= \
+	BL32=$(OPTEE_OS_BIN) \
+	BL33=$(EDK2_BIN) \
+	ARM_TSP_RAM_LOCATION=tdram \
+	PLAT=qemu \
+	DEBUG=0 \
+	LOG_LEVEL=50 \
+	ERROR_DEPRECATED=1 \
+	BL32_RAM_LOCATION=tdram \
+	SPD=opteed
+
+arm-tf: optee-os edk2
+	$(ARM_TF_EXPORTS) $(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) all fip
+
+arm-tf-clean:
+	$(ARM_TF_EXPORTS) $(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) clean
+
+# FIXME: This is just too rough, we should build this just as we're doing for
+#        FVP.
+edk2:
+ifeq ("$(wildcard $(EDK2_BIN))","")
+	mkdir -p $(EDK2_PATH)
+	wget -O $(EDK2_BIN) \
+		http://snapshots.linaro.org/components/kernel/leg-virt-tianocore-edk2-upstream/989/QEMU-KERNEL-AARCH64/RELEASE_GCC49/QEMU_EFI.fd
+endif
+	mkdir -p $(ARM_TF_PATH)/build/qemu/release
+	ln -sf $(OPTEE_OS_BIN) $(ARM_TF_PATH)/build/qemu/release/bl32.bin
```

Comment:
minor: If user calls `make edk2` directly and `optee_os` hasn't been built, then this is a broken link, so maybe add a dependency for it: `edk2: optee-os`? Should be temporary only until edk2 is built from source since edk2 shouldn't really depend on optee-os anyway.